taskcluster/focus_android_taskgraph/transforms/signing.py

```diff
@@ -0,0 +1,91 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Apply some defaults and minor modifications to the jobs defined in the build
+kind.
+"""
+
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.schema import resolve_keyed_by
+
+
+transforms = TransformSequence()
+
+
+@transforms.add
+def resolve_keys(config, tasks):
+    for task in tasks:
+        for key in ("run-on-tasks-for", "signing-format", "notify"):
+            resolve_keyed_by(
+                task,
+                key,
+                item_name=task["name"],
+                **{
+                    'build-type': task["attributes"]["build-type"],
+                    'level': config.params["level"],
+                    'tasks-for': config.params["tasks_for"],
+                }
+            )
+        yield task
+
+
+@transforms.add
+def set_worker_type(config, tasks):
+    for task in tasks:
+        worker_type = "dep-signing"
+        if (
+            str(config.params["level"]) == "3"
+            and task["attributes"]["build-type"]
+            in ("nightly", "beta", "release", "android-test-nightly", "beta-mozillaonline", "release-mozillaonline")
+            and config.params["tasks_for"] in ("cron", "github-release", "action")
+        ):
+            worker_type = "signing"
+        task["worker-type"] = worker_type
+        yield task
+
+
+@transforms.add
+def set_signing_type(config, tasks):
+    for task in tasks:
+        signing_type = "dep-signing"
+        if (
+            str(config.params["level"]) == "3"
+            and config.params["tasks_for"] in ("cron", "github-release", "action")
+        ):
+            if task["attributes"]["build-type"] in ("beta", "release"):
+                signing_type = "fennec-production-signing"
+            elif task["attributes"]["build-type"] in ("nightly", "android-test-nightly", "beta-mozillaonline", "release-mozillaonline"):
```

Comment:
I'm a little bit surprised we're signing test builds with real certs, but Fenix is already doing it, so there's probably a good reason....

Reply:
I don't know.. this was just copy / pasted over from Fenix. Might be better to start without it and then add it later if it turns our it's necessary? Also I guess we can remove these `mozillaonline` values?

Reply:
On closer look, none of these (except `nightly`) are even valid build types in `focus`.. I'll remove them.

Reply:
Yes, actually - the mozillaonline ones can defiintely be removed...